torch/distributed/_sharded_tensor/api.py

```diff
@@ -235,7 +283,7 @@ def _init_rpc(self):
         torch.futures.wait_all(futs)
 
         # Barrier for all RPCs to finish on all ranks.
-        rpc.api._barrier(rpc_workers)
+        rpc.api._all_gather(None)
```

Comment:
why not a barrier anymore?

Reply:
As part of this PR, I'm removing the requirement for sub process groups where you need to do something like:

```
if self.rank >= 2:
   sharded_tensor.empty(....)
```

Its probably cleaner if all ranks just call the sharded_tensor constructor and ranks which are not part of the process group just create an empty ShardedTensor. `_barrier` doesn't support a mode where `rpc_workers` is not specified, that is why I had to use `_all_gather` instead.